enclave-runtime/src/attestation.rs

```diff
@@ -191,6 +191,105 @@ fn generate_dcap_ra_extrinsic_internal(
 	Ok(extrinsic[0].clone())
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn generate_dcap_ra_quote(
+	skip_ra: c_int,
+	quoting_enclave_target_info: &sgx_target_info_t,
+	quote_size: u32,
+	dcap_quote_p: *mut u8,
+	dcap_quote_size: u32,
+) -> sgx_status_t {
+	if dcap_quote_p.is_null() {
+		return sgx_status_t::SGX_ERROR_INVALID_PARAMETER
+	}
+	let dcap_quote = match generate_dcap_ra_quote_internal(
+		skip_ra == 1,
+		quoting_enclave_target_info,
+		quote_size,
+	) {
+		Ok(dcap_quote) => dcap_quote,
+		Err(e) => return e.into(),
+	};
+
+	let dcap_quote_slice = slice::from_raw_parts_mut(dcap_quote_p, dcap_quote_size as usize);
+
+	if let Err(e) = write_slice_and_whitespace_pad(dcap_quote_slice, dcap_quote) {
+		return EnclaveError::Other(Box::new(e)).into()
+	};
+
+	sgx_status_t::SGX_SUCCESS
+}
+
+pub fn generate_dcap_ra_quote_internal(
+	skip_ra: bool,
+	quoting_enclave_target_info: &sgx_target_info_t,
+	quote_size: u32,
+) -> EnclaveResult<Vec<u8>> {
+	let attestation_handler = GLOBAL_ATTESTATION_HANDLER_COMPONENT.get()?;
+
+	let (_, dcap_quote) = attestation_handler.generate_dcap_ra_cert(
+		quoting_enclave_target_info,
+		quote_size,
+		skip_ra,
+	)?;
+
+	Ok(dcap_quote)
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn generate_dcap_ra_extrinsic_with_quote(
+	w_url: *const u8,
+	w_url_size: u32,
+	quote: *const u8,
+	quote_size: u32,
+	unchecked_extrinsic: *mut u8,
+	unchecked_extrinsic_size: u32,
+) -> sgx_status_t {
+	if w_url.is_null() || unchecked_extrinsic.is_null() {
+		return sgx_status_t::SGX_ERROR_INVALID_PARAMETER
+	}
+	let mut url_slice = slice::from_raw_parts(w_url, w_url_size as usize);
+	println!("url_slice is: {:#?}", &url_slice);
+
+	let url = String::decode(&mut url_slice).expect("Could not decode url slice to a valid String");
+	println!("url is: {:#?}", &url);
```

Comment:
I would say we degrade this to debug and make the log more expressive or we remove it.

Reply:
That was a leftover, removed them in 43476fd89f3302a50bd4f9d0e4d309f9f11c0d7f.